python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -704,6 +704,25 @@ def func(pdf):
 
         return self.rdd.map(lambda x: func(x)).first()
 
+    def merge(self, right, how="inner", on=None, **kwargs):
+        """
+        Merge two SparkXShards into a single SparkXShards with a database-style join.
+
+        :param right: The other SparkXShards to be merged.
+        :param how: Type of merge. 'left', 'right', 'outer' or 'inner'. Default is 'inner'.
+        :param on: Column name(s) to join on.
+        :return: A new merged SparkXShards.
+        """   
+        from bigdl.orca.data.utils import spark_df_to_pd_sparkxshards
+        invalidInputError(isinstance(right, SparkXShards), "right should be a SparkXShards")
+
+        left_df = self.to_spark_df()
+        right_df = right.to_spark_df()
+        merged = left_df.join(right_df, on=on, how=how)
+        merged = merged.coalesce(min(merged.count(), merged.rdd.getNumPartitions()))
```

Comment:
@dding3 is this correct?

Reply:
Please change it to `merged.coalesce(left_df.rdd.getNumPartitions())`